flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java

```diff
@@ -1061,8 +1064,12 @@ private static String getDefaultName() {
 	 * @return The execution environment of the context in which the program is executed.
 	 */
 	public static ExecutionEnvironment getExecutionEnvironment() {
+
 		return contextEnvironmentFactory == null ?
-				createLocalEnvironment() : contextEnvironmentFactory.createExecutionEnvironment();
+				(contextEnvironmentFactoryThreadLocal.get() == null ?
```

Comment:
I would suggest to first check the thread local variable and then fall back to the static variable. That way, it would not be possible to override the value of an `ExecutionEnvironment` which has been set explicitly by from another thread.